client/app/bundles/course/leaderboard/components/tables/GroupLeaderboardAchievementsTable.tsx

```diff
@@ -0,0 +1,133 @@
+import { defineMessages, FormattedMessage } from 'react-intl';
+import { Avatar, Tooltip, Link, Box, AvatarGroup } from '@mui/material';
+import DataTable from 'lib/components/DataTable';
+import { GroupLeaderboardAchievement } from 'types/course/leaderboard';
+import { FC, memo } from 'react';
+
+interface Props {
+  data: GroupLeaderboardAchievement[];
+}
+
+const translations = defineMessages({
+  title: {
+    id: 'course.leaderboard.containers.tables.LeaderboardAchievementsTable.title',
+    defaultMessage: 'By Achievement',
+  },
+});
+
+const styles = {
+  title: {
+    flexDirection: 'column',
+    textAlign: 'right',
+    fontSize: 20,
+  },
+  link: {
+    textDecoration: 'none',
+  },
+  avatarGroup: {
+    justifyContent: 'left',
+    '& .MuiAvatar-root': {
+      width: 30,
+      height: 30,
+      marginLeft: '0.1px',
+    },
+  },
+  avatar: {
+    maxWidth: '250px',
+    wordWrap: 'break-word',
+    display: 'inline-flex',
+    alignItems: 'center',
+    minWidth: '150px',
+  },
+};
+
+const GroupLeaderboardAchievementsTable: FC<Props> = (props: Props) => {
+  const data = props.data;
+
+  const columns: any = [
```

Comment:
Can we type this out? I think there is this TableColumns type - you can add more to this type if needed